packages/@ember/array/index.ts

```diff
@@ -515,7 +519,8 @@ interface EmberArray<T> extends Enumerable {
     @return {Array} The mapped array.
     @public
   */
-  getEach<K extends string>(key: K): NativeArray<Value<T, K>>;
+  getEach<K extends keyof T>(key: K): NativeArray<T[K]>;
+  getEach<K extends CompoundPropertyKey<T>>(key: K): NativeArray<unknown>;
```

Comment:
Instead of introducing additional type shenanigans here, since it ultimately just becomes `NativeArray<unknown>`, we can just use `string`. (This is how we do it with the overloads in Ember and Ember Data).

```suggestion
  getEach(key: string): NativeArray<unknown>;
```

Reply:
Resolved.